services/wallet/collectibles/collection_data_db.go

```diff
@@ -130,6 +155,13 @@ func setCollectionsData(creator sqlite.StatementCreator, collections []thirdpart
 			if err != nil {
 				return err
 			}
+
+			if c.Provider == alchemy.AlchemyID && !idExists(creator, c.ID) {
```

Comment:
If the Socials field were a pointer you can just check for non-nil here instead of this provider check. Also, no need to check if it's already in the DB, just replace it with whatever new info we got.